src/Catalyst.Core.Modules.Keystore.Tests/IntegrationTests/KeyApiTest.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            const string name = "net-engine-test-create";
+            var key = await _keyStoreService.CreateAsync(name, "rsa", 512);
+            try
+            {
+                Assert.NotNull(key);
+                Assert.NotNull(key.Id);
+                Assert.Equal(name, key.Name);
+
+                var keys = await _keyStoreService.ListAsync();
+                var clone = keys.Single(k => k.Name == name);
+                Assert.Equal(key.Name, clone.Name);
+                Assert.Equal(key.Id, clone.Id);
+            }
+            finally
+            {
+                await _keyStoreService.RemoveAsync(name);
+            }
+        }
+        
+        [Fact]
+        public async Task Remove_Key()
+        {
+            const string name = "net-engine-test-remove";
+            var key = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+            var keys = await _keyStoreService.ListAsync();
+            var clone = keys.Single(k => k.Name == name);
+            Assert.NotNull(clone);
+
+            var removed = await _keyStoreService.RemoveAsync(name);
+            Assert.NotNull(removed);
+            Assert.Equal(key.Name, removed.Name);
+            Assert.Equal(key.Id, removed.Id);
+
+            keys = await _keyStoreService.ListAsync();
+            Assert.False(keys.Any(k => k.Name == name));
+        }
+
+        [Fact]
+        public async Task Rename_Key()
+        {
+            const string name = "net-engine-test-rename0";
+            const string newName = "net-engine-test-rename1";
+
+            await _keyStoreService.RemoveAsync(name);
+            await _keyStoreService.RemoveAsync(newName);
+            var key = await _keyStoreService.CreateAsync(name, "ed25519", 0);
+            var renamed = await _keyStoreService.RenameAsync(name, newName);
+            Assert.Equal(key.Id, renamed.Id);
+            Assert.Equal(newName, renamed.Name);
+
+            var keys = await _keyStoreService.ListAsync();
+            var enumerable = keys as IKey[] ?? keys.ToArray();
+            Assert.True(enumerable.Any(k => k.Name == newName));
+            Assert.False(enumerable.Any(k => k.Name == name));
+        }
+
+        [Fact]
+        public async Task Remove_Unknown_Key()
+        {
+            const string name = "net-engine-test-remove-unknown";
+
+            var removed = await _keyStoreService.RemoveAsync(name);
+            Assert.Null(removed);
+        }
+
+        [Fact]
+        public async Task Rename_Unknown_Key()
+        {
+            const string name = "net-engine-test-rename-unknown";
+
+            var renamed = await _keyStoreService.RenameAsync(name, "foobar");
+            Assert.Null(renamed);
+        }
+
+        [Fact]
+        public void Create_Unknown_KeyType()
+        {
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = _keyStoreService.CreateAsync("unknown", "unknown", 0).Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)